src/dropdown/dropdown.js

```diff
@@ -187,17 +187,30 @@ angular.module('ui.bootstrap.dropdown', ['ui.bootstrap.position'])
     }
   };
 
+  function toggleDropdownUp() {
```

Comment:
@RobJacobs 
Indeed it works fine with placement set to auto, but I noticed that width and height of my targetElement inside the $position.positionElements are 0.

`// need to read from prop to support tests.
        var targetWidth = angular.isDefined(targetElem.offsetWidth) ? targetElem.offsetWidth : targetElem.prop('offsetWidth');
        var targetHeight = angular.isDefined(targetElem.offsetHeight) ? targetElem.offsetHeight : targetElem.prop('offsetHeight');`

Why not use the outerWidth and outerHeight?